invenio_app_ils/circulation/api.py

```diff
@@ -210,6 +212,14 @@ def checkout_loan(
         loan, **dict(params, trigger="checkout")
     )
 
+    current_end_date = arrow.get(loan["end_date"]).date()
+    new_end_date = next_open_after(loan["pickup_location_pid"], current_end_date)
+    if new_end_date and new_end_date != current_end_date:
+        loan.update(end_date=new_end_date.isoformat())
+        loan.commit()
+        db.session.commit()
+        current_app_ils.location_indexer.index(loan)
```

Comment:
Instead of updating/reindexing the record twice, I believe it would be better to define a new `Transition` class that inherits from `ToItemOnLoan` and executes this procedure in the `before` method (after calling `super().before`).